msteams-platform/messaging-extensions/high-quality-message-extension.md

```diff
@@ -0,0 +1,511 @@
+          "type": "query",
+          "title": "Tasks",
+          "description": "Search for high priority tasks related to Northwind that are due tomorrow.",
+          "initialRun": true,
+          "fetchTask": false,
+          "context": [
+            "commandBox",
+            "compose",
+            "message"
+          ],
+```
+
+# [Surveys](#tab/surveys)
+
+**Description**: Search for surveys, drafts, and results with keywords or number of respondents.
+
+**Command description example:**
+
+```json
+"commands": [
+        {
+          "id": "Search",
+          "type": "query",
+          "title": "Survey",
+          "description": "Search for surveys, drafts, and results with keywords or number of respondents.",
+          "initialRun": true,
+          "fetchTask": false,
+          "context": [
+            "commandBox",
+            "compose",
+            "message"
+          ],
+```
+
+# [CRM](#tab/crm)
+
+**Description**: Through CRM plugin, find qualified, unqualified, and quoted leads of clients and customers.
+
+**Command description example:**
+
+```json
+"commands": [
+        {
+          "id": "Search",
+          "type": "query",
+          "title": "CRM",
+          "description": "Through CRM plugin, find qualified, unqualified, and quoted leads of clients and customers.",
+          "initialRun": true,
+          "fetchTask": false,
+          "context": [
+            "commandBox",
+            "compose",
+            "message"
+          ],
+```
+
+# [General](#tab/general)
+
+**Description**: Find number of stocks or listed equities using keyworks, key ratios, index, and so on.
+
+**Command description example:**
+
+```json
+"commands": [
+        {
+          "id": "Search",
+          "type": "query",
+          "title": "General",
+          "description": "Find number of stocks or listed equities using keyworks, key ratios, index, and so on.",
+          "initialRun": true,
+          "fetchTask": false,
+          "context": [
+            "commandBox",
+            "compose",
+            "message"
+          ],
+```
+
+---
+
+### Parameter description
+
+Each parameter supports five attributes and one attribute must be visible in the message extension search bar. A parameter must have a good description, which must contain a combination of acceptable parameters, Enums, acronyms, and output format.
+
+A good parameter description explains the requirements of the system in a natural language with output format. The following are a few examples of basic and advanced search requests for different categories:
+
+# [Tasks](#tab/tasks)
+
+Basic search: Search for tasks related to Northwind.</br>
+Advanced search: Search for high priority tasks related to Northwind that are due tomorrow.
+
+**Parameter description example:**
+
+```json
+"parameters": [
+    {
+        "name": "Name",
+        "title": "Project or Task Name",
+        "description": "Send project name or task name as keyword",
```

Comment:
Per Abram, param descriptions shouldn't include verbs (only noun describing parameter itself).

"description": "Project name or task name as keyword"

Reply:
(Remove other instances of "Send" below)

Reply:
Remove instances of "Send in the examples.